doc/introduction/remote.rst

```diff
@@ -0,0 +1,511 @@
+         -0.18224807+0.30281836j,  0.26930442-0.04644871j, -0.46045639-0.55359506j, -0.0737605-0.52580999j
+         0.19903677-0.43076659j, -0.50320649-0.44750373j, -0.01617065-0.52755812j, -0.19729219+0.06200712j
+
+    # Initial states are two-mode squeezed states
+    S2gate(1.0, 0.0) | [0, 4]
+    S2gate(1.0, 0.0) | [1, 5]
+    S2gate(1.0, 0.0) | [2, 6]
+    S2gate(1.0, 0.0) | [3, 7]
+
+    # Apply the unitary matrix above to
+    # the first pair of modes, as well
+    # as a beamsplitter
+    Interferometer(U) | [0, 1, 2, 3]
+    BSgate(0.543, -0.123) | [0, 1]
+
+    # Duplicate the above unitary for
+    # the second pair of modes
+    Interferometer(U) | [4, 5, 6, 7]
+    BSgate(0.543, -0.123) | [4, 5]
+
+    # Perform a PNR measurement in the Fock basis
+    MeasureFock() | [0, 1, 2, 3, 4, 5, 6, 7]
+
+
+**Note:** You may use :func:`~.random_interferometer` to generate arbitrary random unitaries.
+
+This program will execute following the same steps as above; :class:`~.RemoteEngine` will automatically
+compile the program to match the layout of the chip.
+
+You may wish to view the compiled program; this can be easily done in Python using
+the ``Program.compile`` method:
+
+
+>>> from strawberryfields import RemoteEngine
+>>> from strawberryfields.io import load
+>>> prog = load("test.xbb")
+>>> prog = prog.compile("chip2")
+>>> prog.print()
+S2gate(1, 0) | (q[0], q[4])
+S2gate(1, 0) | (q[3], q[7])
+S2gate(1, 0) | (q[2], q[6])
+MZgate(1.573, 4.368) | (q[2], q[3])
+MZgate(1.573, 4.368) | (q[6], q[7])
+S2gate(1, 0) | (q[1], q[5])
+MZgate(1.228, 5.006) | (q[0], q[1])
+MZgate(4.414, 3.859) | (q[1], q[2])
+MZgate(2.98, 3.316) | (q[2], q[3])
+Rgate(-0.7501) | (q[3])
+MZgate(5.397, 5.494) | (q[0], q[1])
+MZgate(5.152, 4.891) | (q[1], q[2])
+Rgate(2.544) | (q[2])
+MZgate(1.228, 5.006) | (q[4], q[5])
+MZgate(4.414, 3.859) | (q[5], q[6])
+MZgate(2.98, 3.316) | (q[6], q[7])
+Rgate(-0.7501) | (q[7])
+MZgate(5.397, 5.494) | (q[4], q[5])
+MZgate(5.152, 4.891) | (q[5], q[6])
+Rgate(2.544) | (q[6])
+Rgate(-1.173) | (q[1])
+Rgate(1.902) | (q[4])
+Rgate(1.902) | (q[0])
+Rgate(-1.173) | (q[5])
+MeasureFock | (q[0], q[1], q[2], q[3], q[4], q[5], q[6], q[7])
+
+and even saved as a new Blackbird script using the :func:`~.io.save` function:
+
+>>> from strawberryfields.io import save
+>>> save("test_compiled.xbb", prog)
+
+
+Tips and tricks
+---------------
+
+.. code-block:: python3
+
+    from strawberryfields.utils import operation
+
+We can define an operation to make it easier to apply the same unitary
+to both signal and idler modes.
+
+.. code-block:: python3
+
+    @operation(4)
+    def unitary(q):
+        ops.Interferometer(U) | q
+        ops.BSgate(0.543, 0.123) | (q[2], q[0])
+
+    prog = sf.Program(8)
+
+    with prog.context as q:
+        ops.S2gate(1.0) | (q[0], q[4])
+        ops.S2gate(1.0) | (q[1], q[5])
+        ops.S2gate(1.0) | (q[2], q[6])
+        ops.S2gate(1.0) | (q[3], q[7])
+
+        unitary() | q[:4]
```

Comment:
Will it be confusing to users if the call to `unitary` here does not match the signature above?